enhancements/update/manifest-install-levels.md

```diff
@@ -0,0 +1,166 @@
+---
+title: manifest-install-levels
+authors:
+  - "@wking"
+reviewers:
+  - "@LalatenduMohanty"
+approvers:
+  - TBD
+creation-date: 2020-09-12
+last-updated: 2020-09-12
+status: implementable
+---
+
+# Manifest Install Levels
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+[Several operators][extra-manifests] pass manifests [like these][extra-manifests-example] to the cluster-version operator which are nice-to-have, [create-only][] defaults, but not critical to bootstrap completion.
+That's fine, and makes life in the post-install clusters easier.
+However, when a user feeds the installer their own alternative content, [cluster-bootstrap][] will [race][] the cluster-version operator to push the content into the cluster.
+Sometimes cluster-bootstrap wins, and the user-provided manifest ends up in-cluster.
+But sometimes the cluster-version operator wins, and the user-provided content is forgotten when the create-only default ends up in-cluster.
+This enhancement adds a new enhancement to delay non-critical manifests to later in the installation, ensuring the cluster-version operator will always lose the race and cluster-bootstrap will push the user-provided manifest into the cluster.
+
+## Motivation
+
+### Goals
+
+* Respect manifests that users feed the installer, even if they target a resource that is also backed by a cluster-version operator, create-only manifest.
```

Comment:
do we really want to support those clusters? My gut feeling is a strong no. This basically opens the door to customers running things we have never seen and never tested. This is very concerning. cc @derekwaynecarr @smarterclayton

Reply:
These manifests are "things you change after install" which includes "things that may be necessary for bootstrap, like a custom networking impl".  We fully support customers changing config after install and customers using custom operators at install time.  I think the word "manifests" appears confusing when conflated with payload manifests - we should make clear that these manifests are "valid configuration target manifests like our config api".

Reply:
As per slack chat we should precisely define the categories in the proposal with clear terminology and then use it in the design:

```
manifests-in-payload
manifests-in-payload-that-are-supported-for-customer-override
manifests-in-payload-that-are-not-supported-for-customer-override
customer-provided-manifests-for-config
customer-provided-manifests-for-additional-operators-required-to-bootstrap
```

Reply:
I've added a terminology section to the enhancement, and I've also added an explicit guard to restrict this new annotation to manifests which already set the create-only manifest.

```console
$ oc adm release extract --to manifests quay.io/openshift-release-dev/ocp-release:4.6.0-fc.7-x86_64
Extracted release payload from digest sha256:99fdfbd1af951f1b5cf2e2aa1a79b6c2ec565950817ffcaa8bac54014d1852dc created at 2020-09-18T09:31:51Z
$ grep -r /create-only manifests
manifests/0000_05_config-operator_02_proxy.cr.yaml:    release.openshift.io/create-only: "true"
manifests/0000_05_config-operator_02_build.cr.yaml:    release.openshift.io/create-only: "true"
manifests/0000_05_config-operator_02_ingress.cr.yaml:    release.openshift.io/create-only: "true"
manifests/0000_50_console-operator_04-rbac-rolebinding-cluster.yaml:    "release.openshift.io/create-only": 'true'
...
```

I am not clear on how breaking down customer-provided manifests, or any manifests applied via cluster-bootstrap, into subcategories is useful.  This enhancement proposal is about keeping optional release image manifests out of the cluster until cluster-bootstrap is done.  It is not about changing anything about how cluster-bootstrap operates.

I am not clear on the importance of distinguishing based on support for "customer overrides" (do you mean "customer configuration"?).  I would expect release image manifests which support customer configuration to set the create-only annotation, otherwise the CVO would just stomp the customer's changes.

It's possible that there are create-only release image manifests which are still not customer-managed on day 2.  For example, perhaps the CVO creates the manifest, but then a second-level operator picks up management of that manifest later?  I am not aware of any such manifests.  But if there's something similar that drives a need to distinguish some subset of create-only release image manifests where the new `release.openshift.io/install-level` should not apply, can we give a specific example from 4.6.0-fc.7?